src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java

```diff
@@ -187,88 +187,105 @@ public void check(GHIssueComment comment) {
-    //         the last update time.
-    private void updatePR(GHPullRequest ghpr, GHIssueComment comment, boolean isWebhook) {
+    // 1. With webhooks + new trigger/PR initialization -
+    // This could happen if a new job was added, new trigger was enabled, or if Jenkins
+    // was restarted. In this case, our view of the PR is out of date. We need to
+    // compare hashes and check the comments going back to when the last update was (which could be
+    // when the PR was created).
+    // 2. With webhooks + new comment/PR update - This is "normal" operation. In these
+    // cases, we only need to process the comment that was just added, or compare hashes with
+    // the updated PR info (for instance, if someone changes a title of a PR it shouldn't trigger.
+    // We do NOT need to pull the comment info, since we will have gotten or will get
+    // each comment.
+    // 3. Without webhooks - In this case, we will always check comments and hashes until
+    // the last update time.
+    private void updatePR(GHPullRequest ghpr,
+                          GHIssueComment comment,
+                          boolean isWebhook) {
         // Get the updated time
         try {
-            Date lastUpdateTime = updated;
-            Date updatedDate = comment != null ? comment.getUpdatedAt() : ghpr.getUpdatedAt();
-            // Don't log unless it was actually updated
-            if (updated == null || updated.compareTo(updatedDate) < 0) {
-                String user = comment != null ? comment.getUser().getName(): ghpr.getUser().getName();
-                logger.log(Level.INFO, "Pull request #{0} was updated/initialized on {1} at {2} by {3} ({4})", new Object[] { this.id, this.repo.getName(), updatedDate, user,
-                    comment != null ? "comment" : "PR update"});
-            }
-        
-            // Update the PR object with the new pull request object if
-            // it is non-null.  getPullRequest will then avoid another
-            // GH API call.
-            if (ghpr != null) {
-                setPullRequest(ghpr);
-            }
-            
-            // Grab the pull request for use in this method (in case we came in through the comment path)
-            GHPullRequest pullRequest = getPullRequest();
-            
             synchronized (this) {
-                boolean wasUpdated = setUpdated(updatedDate);
-                
+
+                // Update the PR object with the new pull request object if
+                // it is non-null. getPullRequest will then avoid another
+                // GH API call.
+                if (ghpr != null) {
+                    setPullRequest(ghpr);
+                }
+
+                Date lastUpdateTime = updated;
+                Date updatedDate = comment != null ? comment.getUpdatedAt() : ghpr.getUpdatedAt();
+                boolean wasUpdated = isUpdated(getPullRequest(), updatedDate);
```

Comment:
Why is the hash comparison necessary here? If the PR or comment was updated, we will get a newer updated date, which should lead to the hash comparison below.  It seems like the old code was correct.  Instead,we now compare twice.